combat-trainer.lic

```diff
@@ -3981,6 +3989,14 @@ class GameState
     !@clean_up_step.nil?
   end
 
+  def can_engage?
+    return false if @innocence_mode
+    return false if retreating?
+    return false if npcs.empty?
+    return false if danger
```

Comment:
Not certain it makes sense for this `danger` check to be here too. Will we always be `retreating?` if we're in danger?